### DISCUSSION TITLE
How to access router from model?

### DISCUSSION THREAD

**Discussion by Krzysiaczek** [phalcon/cphalcon#15435] [Q&A]
I am trying to access the method name from a Model to modify the query logic but it looks like it's not visible from here.
`$this->router->getActionName()` works perfect in a controller or a view but when used in a model it's not visible  
`Access to undefined property App\Models\Activities::router `

When trying to call it with a domain name `\Phalcon\Mvc\Router::getActionName()`
`Fatal error: Uncaught Error: Non-static method Phalcon\Mvc\Router::getActionName() cannot be called statically`

Any advice on how to fix it?

**Comment by wurst-hans** [comment]
Try to use

`$method = $this->container->get('router')->getActionName();`

**Reply by Krzysiaczek-at-MasAndPas** [reply]
Thanks for the advise. This method doesn't throw any errors but returns NULL.

**Comment by quasipickle** [comment]
Following separation of concerns, Models should have no idea about the controller or action that has been invoked.  That introduces a dependency that could cause trouble down the road.

I think a better approach would be to have your model method accept an optional parameter. The action can then pass an argument to the model method when, and only if, it's necessary.

Perhaps seeing the code of the method might help.
